src/java.base/unix/native/libjava/ProcessImpl_md.c

```diff
@@ -559,7 +559,9 @@ spawnChild(JNIEnv *env, jobject process, ChildStuff *c, const char *helperpath)
     }
     offset = copystrings(buf, 0, &c->argv[0]);
     offset = copystrings(buf, offset, &c->envv[0]);
-    memcpy(buf+offset, c->pdir, sp.dirlen);
+    if (c->pdir != NULL) {
+        memcpy(buf+offset, c->pdir, sp.dirlen);
+    }
     offset += sp.dirlen;
```

Comment:
I'd be inclined to check sp.dirlen > 0 in the `if` and move the offset += inside too. Like:
```suggestion
    if (sp.dirlen > 0 && c->pdir != NULL) {
        memcpy(buf+offset, c->pdir, sp.dirlen);
        offset += sp.dirlen;
    }
```
The behavior is correct either way.